src/avram/model.cr

```diff
@@ -8,82 +8,27 @@ abstract class Avram::Model
   include Avram::Polymorphic
   include Avram::SchemaEnforcer
 
-  abstract def id
-
   macro inherited
     COLUMNS = [] of Nil # types are not checked in macros
     ASSOCIATIONS = [] of Nil # types are not checked in macros
   end
 
-  def_equals id, model_name
+  def self.primary_key_name : Symbol?
+    nil
+  end
 
   def model_name
     self.class.name
   end
 
-  def to_param
-    id.to_s
-  end
-
-  # Reload the model with the latest information from the database
-  #
-  # This method will return a new model instance with the
-  # latest data from the database. Note that this does
-  # **not** change the original instance, so you may need to
-  # assign the result to a variable or work directly with the return value.
-  #
-  # Example:
-  #
-  # ```crystal
-  # user = SaveUser.create!(name: "Original")
-  # SaveUser.update!(user, name: "Updated")
-  #
-  # # Will be "Original"
-  # user.name
-  # # Will return "Updated"
-  # user.reload.name # Will be "Updated"
-  # # Will still be "Original" since the 'user' is the same model instance.
-  # user.name
-  #
-  # Instead re-assign the variable. Now 'name' will return "Updated" since
-  # 'user' references the reloaded model.
-  # user = user.reload
-  # user.name
-  # ```
+  # Refer to `PrimaryKeyMethods#reload`
   def reload : self
-    base_query_class.find(id)
+    raise Avram::UnsupportedQueryError.new("Unable to call Avram::Model#reload on #{self.class.name} because no primary key has been defined.")
```

Comment:
What do you think about making this a compile-time error? I think for it to work, the definitions in `PrimaryKeyMethods` need to be inside of the macro `included` 🤔 

```crystal
{% raise "Unable to call Avram::Model#reload on {{ @type.id }} because no primary key has been defined." %}
```

Reply:
Great idea! I had no idea you could do this. Now if only I could get my brain to remember it 🤦‍♂️🤦‍♂️🤦‍♂️